src/app/shared/components/course-card/course-card.component.ts

```diff
@@ -1,8 +1,27 @@
-import { Component } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { mapAuthorIdsToNames } from '@app/utilities/author-utils';
+import { mockedAuthorsList } from '@shared/mocks/mocks';
+import { Course } from '@features/courses/course';
 
 @Component({
   selector: 'app-course-card',
   templateUrl: './course-card.component.html',
   styleUrls: ['./course-card.component.scss'],
 })
-export class CourseCardComponent {}
+export class CourseCardComponent implements OnInit {
+  @Input() course!: Course;
+  @Input() isEditable = true;
+  authorsNames: string[] = [];
+
+  ngOnInit() {
+    this.authorsNames = mapAuthorIdsToNames(
```

Comment:
As you move the data read from here, mapping will have to move, too or even be eliminated.

Reply:
I moved the whole mapping logic to the Courses component and passed down to the course-card component.